hello_world_on_chain/test/user.ts

```diff
@@ -1,79 +1,92 @@
+// noinspection DuplicatedCode
+
 import hre from "hardhat";
 import {loadFixture} from "@nomicfoundation/hardhat-toolbox/network-helpers";
 import {expect} from "chai";
 import {ethers} from "ethers";
+import fs from "fs";
+import path from "path";
+import {User} from "../typechain-types";
 
-const fs = require("fs")
-const path = require("path")
+type EVMLinkData = {
+    publicKey: string;
+    transitiveAttestation: string;
+};
 
-function loadEVMLinkData(jsonPath: string) {
+function loadEVMLinkData(jsonPath: string): EVMLinkData {
     try {
-        const dir = path.resolve( __dirname, jsonPath);
-        const file = fs.readFileSync(dir);
+        const dir: string = path.resolve( __dirname, jsonPath);
+        const file: string = fs.readFileSync(dir, "utf8");
 
-        const data = JSON.parse(file);
+        const data: any = JSON.parse(file);
```

Comment:
I'll only write this once: you'll notice that I've added explicit types to all variables and function definitions. I wasn't getting warnings per-se, but Goland was adding "ghost" script everywhere telling me what all the inferred types were. I found that more annoying than explicitly having the types written out in code and figured it would be helpful to those whose IDE does not show them inferred types. Plus, it's called "type" script so we might as well write them, right?